solutions/account-infrastructure-base/main.tf

```diff
@@ -21,18 +21,53 @@ module "account_infrastructure_base" {
+  activity_tracker_route_name          = var.activity_tracker_route_name == null ? try("${var.prefix}-cos-route", "cos-route") : var.activity_tracker_route_name
+  cos_bucket_management_endpoint_type  = var.cos_bucket_management_endpoint_type
+  kms_key_crn                          = var.kms_key_crn
+  resource_tags                        = var.resource_tags
+  cos_plan                             = var.cos_plan
+  cos_instance_access_tags             = var.cos_instance_access_tags
+  cos_bucket_access_tags               = var.cos_bucket_access_tags
+  cos_bucket_expire_enabled            = var.cos_bucket_expire_enabled
+  cos_bucket_expire_days               = var.cos_bucket_expire_days
+  cos_bucket_object_versioning_enabled = var.cos_bucket_object_versioning_enabled
+  cos_bucket_storage_class             = var.cos_bucket_storage_class
+  cos_bucket_archive_enabled           = var.cos_bucket_archive_enabled
+  cos_bucket_archive_days              = var.cos_bucket_archive_days
+  cos_bucket_archive_type              = var.cos_bucket_archive_type
+  cos_bucket_retention_enabled         = var.cos_bucket_retention_enabled
+  cos_bucket_retention_default         = var.cos_bucket_retention_default
+  cos_bucket_retention_maximum         = var.cos_bucket_retention_maximum
+  cos_bucket_retention_minimum         = var.cos_bucket_retention_minimum
+  cos_bucket_retention_permanent       = var.cos_bucket_retention_permanent
+  cos_bucket_cbr_rules                 = var.cos_bucket_cbr_rules
+  cos_instance_cbr_rules               = var.cos_instance_cbr_rules
+  skip_atracker_cos_iam_auth_policy    = var.skip_atracker_cos_iam_auth_policy
+  activity_tracker_locations           = var.activity_tracker_locations
 
   # iam account settings
-  allowed_ip_addresses = var.allowed_ip_addresses
+  allowed_ip_addresses         = var.allowed_ip_addresses
+  access_token_expiration      = var.access_token_expiration
+  active_session_timeout       = var.active_session_timeout
+  api_creation                 = var.api_creation
+  enforce_allowed_ip_addresses = var.enforce_allowed_ip_addresses
+  inactive_session_timeout     = var.inactive_session_timeout
+  max_sessions_per_identity    = var.max_sessions_per_identity
+  mfa                          = var.mfa
+  public_access_enabled        = var.public_access_enabled
+  refresh_token_expiration     = var.refresh_token_expiration
+  serviceid_creation           = var.serviceid_creation
+  shell_settings_enabled       = var.shell_settings_enabled
+  user_mfa                     = var.user_mfa
+  user_mfa_reset               = var.user_mfa_reset
 
   # trusted profile
-  trusted_profile_name = try("${var.prefix}-trusted-profile", "projects-trusted-profile")
+  trusted_profile_name               = var.trusted_profile_name == null ? try("${var.prefix}-trusted-profile", "projects-trusted-profile") : var.trusted_profile_name
```

Comment:
try not needed here now

Reply:
Seems as per the logic of try we were using previously it would be like this now - 
* if `trusted_profile_name` is null then try with `prefix` and if `prefix` is null/errored then just name it `projects-trusted-profile`.
* otherwise use `trusted_profile_name`

Reply:
Ah sorry, you are right, prefix didn't show in the diff and I thought it was a requirement